core/src/main/kotlin/plugins/Authentication.kt

```diff
@@ -83,6 +85,17 @@ fun Application.configureAuthentication() {
             validate { credential ->
                 credential.payload.takeIf(this@configureAuthentication::validateJwtPayload)?.let { payload ->
                     val roles = getRoles(payload.subject, keycloakClient, roleCache)
+
+                    // Clear the cache for the user after a write operation as it could have changed the roles.
+                    @Suppress("ComplexCondition")
+                    if (this.request.httpMethod == HttpMethod.Delete ||
```

Comment:
As a quick fix, this is probably okay. But on the long run, I would prefer to have a more focused invalidation when it is actually needed. Also, a test would be nice.

Reply:
Ideally we could find a solution that allows us to remove the cache altogether, like https://github.com/eclipse-apoapsis/ort-server/issues/984#issuecomment-2550639579.